src/op_mode/ipsec.py

```diff
@@ -135,6 +135,225 @@ def _get_formatted_output_sas(sas):
     return output
 
 
+# Connections block
+def _get_vici_connections():
+    from vici import Session as vici_session
+
+    try:
+        session = vici_session()
+    except Exception:
+        raise vyos.opmode.UnconfiguredSubsystem("IPsec not initialized")
+    connections = list(session.list_conns())
+    return connections
+
+
+def _get_convert_data_connections():
+    get_connections = _get_vici_connections()
+    connections = convert_data(get_connections)
+    return connections
+
+
+def _get_parent_sa_proposal(connection_name: str, data: list) -> str:
+    """Get parent SA proposals by connection name
+    if connections not in the 'down' state
+
+    Args:
+        connection_name (str): Connection name
+        data (list): List of current SAs from vici
+
+    Returns:
+        str: Parent SA connection proposal
+             AES_CBC/256/HMAC_SHA2_256_128/MODP_1024
+    """
+    for sa in data:
+        # check if parent SA exist
+        if connection_name not in sa.keys():
+            return '-'
+        encr_alg = sa.get(connection_name, '').get('encr-alg')
+        encr_keysize = sa.get(connection_name, '').get('encr-keysize')
+        integ_alg = sa.get(connection_name, '').get('integ-alg')
+        dh_group = sa.get(connection_name, '').get('dh-group')
+        proposal = f'{encr_alg}/{encr_keysize}/{integ_alg}/{dh_group}'
```

Comment:
I believe this formatting should be done _only_ in human-readable formatted output. The raw output show include the cipher, the key size, and the hash as separate fields. Well, the DH group when applicable, as well, as a separate field.

If we do it like this and send monolithic strings to clients, clients will _parse_ it to display proposal details in GUIs etc. and that particular format will become de facto a public API contact. We most definitely don't want that. ;)

Ideally, we should even separate the cipher and its mode, like this:

```
{
  "cipher": "aes",
  "mode": "cbc",
  "key_size": 256,
  "hash": "HMAC_SHA2_256_128",
  "dh": "MODP_1024"
}
```

We may even want to convert `HMAC_SHA2_256_128` to more conventional names like `SHA-256` (or, maybe, `SHA2-256`), unless there are compelling reasons not to. I have to admit I can't remember what's 128 in that example and I'll need to find out. But I'm sure in an overview in a GUI or similar, people will want to see it as `SHA-256`.

Reply:
So if we want to convert we need all possible values that we want
or just show it as it is.